### PR TITLE
Remove ES7 from examples

### PR DESCRIPTION
**Pull request by rwillrich**
Here is some features I have removed:

- Decorators;
- Bind syntax;
- Class properties;
- Object spread.

### CODE REVIEW

**Comment by quicksnap** [comment]
These examples don't have automated tests AFAIK--did you test them out? =)

**Comment by rwillrich** [comment]
I've tested them manually. Maybe we should open an issue to add test coverage to them.

**Comment by quicksnap** [comment]
I haven't worked much with Travis--how well would it deal with a nested project?

**Comment by rwillrich** [comment]
Don't know. What do you think @gaearon?

**Comment by quicksnap** [comment]
I kind of see how it works.. I'll open an issue on it.

**Comment by gaearon** [comment]
Awesome, thank you.

**Comment by rwillrich** [comment]
You're welcome! :wink:

**Comment by ms88privat** [comment]
Why no ES7 anymore?

**Comment by quicksnap** [comment]
I believe they're being removed from the examples because not everyone is
comfortable using non-standard language features--doesn't make them
suitable for general purpose examples.

**Comment by mindjuice** [comment]
You could have have ES5, ES6 and ES7 versions of the examples like @gaearon did for React DnD. https://gaearon.github.io/react-dnd/

Pick your poison.

**Comment by gaearon** [comment]
I think at this point everybody who knows about ES7 features will use them anyway, so no big deal.

Maintaining three versions of examples is too big a burden for me at this point.

File: examples/counter/components/Counter.js [hunk 1]
```diff
@@ -23,3 +16,12 @@ export default class Counter extends Component {
...
 }
+
+Counter.propTypes = {
+  increment: PropTypes.func.isRequired,
+  incrementIfOdd: PropTypes.func.isRequired,
+  decrement: PropTypes.func.isRequired,
+  counter: PropTypes.number.isRequired
+};
+
+export default Counter;
```
Comment (baddox) [inline examples/counter/components/Counter.js]: I thought it works fine to `export default class Foo` at the top and still add propTypes after the class declaration.

(Forgive me if I'm wrong, I can't test it out right now.)

Comment (knowbody) [inline examples/counter/components/Counter.js]: @baddox I do agree. looks weird but it's correct. maybe send a PR

**Pull request merged**